library/Require.hs

```diff
@@ -24,109 +47,206 @@ data CommandArguments =
 
 instance ParseRecord CommandArguments
 
-findRequires :: IO (Maybe Text)
+
+type LineTagPrepend = LineTag -> Text -> Text
+
+data TransformState = TransformState
+  { _tstLineTagPrepend :: !LineTagPrepend
+  , _tstHostModule     :: !(Maybe ModuleName)
+  , _tstAutorequired   :: !Bool
+  }
+
+makeLenses ''TransformState
```

Comment:
If using Template Haskell is not wanted, I can just write the lenses "by hand" using the `lens` function – if using lenses is ok at all :)